cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -60,35 +56,83 @@ case class DrsPathResolver(drsConfig: DrsConfig, httpClientBuilder: HttpClientBu
     } yield httpResponse
   }
 
-
-  def rawMarthaResponse(drsPath: String, serviceAccount: Option[String] = None): Resource[IO, HttpResponse] = {
-    val httpPost = makeHttpRequestToMartha(drsPath, serviceAccount)
+  def rawMarthaResponse(drsPath: String): Resource[IO, HttpResponse] = {
+    val httpPost = makeHttpRequestToMartha(drsPath)
     executeMarthaRequest(httpPost)
   }
 
   /** *
     * Resolves the DRS path through Martha url provided in the config.
     * Please note, this method returns an IO that would make a synchronous HTTP request to Martha when run.
     */
-  def resolveDrsThroughMartha(drsPath: String, serviceAccount: Option[String] = None): IO[MarthaResponse] = {
-    rawMarthaResponse(drsPath, serviceAccount).use(httpResponseToMarthaResponse)
+  def resolveDrsThroughMartha(drsPath: String): IO[MarthaResponse] = {
+    rawMarthaResponse(drsPath).use(httpResponseToMarthaResponse)
   }
 }
 
-case class DrsConfig(marthaUri: String, marthaRequestJsonTemplate: String)
 
+case class DrsConfig(marthaUri: String, marthaRequestJsonTemplate: String)
 
 case class Url(url: String)
-
 case class ChecksumObject(checksum: String, `type`: String)
-
 case class DrsDataObject(size: Option[Long],
                          checksums: Option[Array[ChecksumObject]],
                          updated: Option[String],
                          urls: Array[Url])
-
 case class DrsObject(data_object: DrsDataObject)
-
 case class SADataObject(data: Json)
 
-case class MarthaResponse(drs: DrsObject, googleServiceAccount: Option[SADataObject])
+case class MarthaV2Response(dos: DrsObject, googleServiceAccount: Option[SADataObject])
+
+case class MarthaResponse(size: Option[Long],
+                          timeUpdated: Option[String],
+                          bucket: Option[String],
+                          name: Option[String],
+                          gsUri: Option[String],
+                          googleServiceAccount: Option[SADataObject],
+                          hashes: Option[Map[String, String]])
+
+object MarthaResponseSupport {
+
+  implicit lazy val urlDecoder: Decoder[Url] = deriveDecoder
+  implicit lazy val checksumDecoder: Decoder[ChecksumObject] = deriveDecoder
+  implicit lazy val dataObjectDecoder: Decoder[DrsDataObject] = deriveDecoder
+  implicit lazy val drsObjectDecoder: Decoder[DrsObject] = deriveDecoder
+  implicit lazy val saDataObjectDecoder: Decoder[SADataObject] = deriveDecoder
+  implicit lazy val marthaV2ResponseDecoder: Decoder[MarthaV2Response] = deriveDecoder
+  implicit lazy val marthaV3ResponseDecoder: Decoder[MarthaResponse] = deriveDecoder
```

Comment:
Changed to `marthaResponseDecoder`; the thinking is that (hopefully) sometime soon, `martha_v2` will be retired, so there will be only one concept of a "Martha" response.

Reply:
Until v4 comes along? I don't think it's necessarily bad to indicate that this is specific to v3 but I'll leave it up to you

Reply:
This might be overly optimistic of me, but I'm hoping that even if/when there is a v4, the Martha response structure won't fundamentally change, so we won't need to support multiple formats.  History will be the judge 🤷 .